pkg/ingester/ingester.go

```diff
@@ -2482,3 +2483,22 @@ func allOutOfBounds(samples []mimirpb.Sample, minValidTime int64) bool {
 	}
 	return true
 }
+
+func (i *Ingester) UserRegistryHandler(w http.ResponseWriter, r *http.Request) {
+	userID, err := tenant.TenantID(r.Context())
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusUnauthorized)
+		return
+	}
+
+	reg := i.tsdbMetrics.regs.GetRegistryForUser(userID)
+	if reg == nil {
+		http.Error(w, "user registry not found", http.StatusNotFound)
+		return
+	}
+
+	promhttp.HandlerFor(reg, promhttp.HandlerOpts{
+		ErrorHandling: promhttp.HTTPErrorOnError,
```

Comment:
Have you considered setting `DisableCompression: true` and then enable compression when we `RegisterRoute()`? Would be consistent with other API endpoints registered by Mimir.

Reply:
I haven't, but sounds like a good idea. Fixed in 8a7e1947bf7f033876031d6264f3ff72647483c4.